exports/dataset.cpp

```diff
@@ -517,6 +538,8 @@ PYBIND11_MODULE(dataset, m) {
       .def_property_readonly("tag", &Variable::tag)
       .def_property("name", [](const Variable &self) { return self.name(); },
                     &Variable::setName)
+      .def_property("unit", [](const Variable &self) { return self.unit(); },
```

Comment:
Not sure why the lambda is needed here, doesn't `&Variable::unit()` work? Same for `&Variable::name()` above?

Reply:
yes, `&Variable::unit()` works, but it doesn't work for the `name()` above. Probably something to do with the not so trivial implementation of `name()` (https://github.com/mantidproject/dataset/blob/master/src/variable.h#L186)